src/main/scala/rps/service/GameService.scala

```diff
@@ -5,9 +5,14 @@ import scala.util.Random
 import rps.model.{GameSummary, Move, Result}
 import Move._, Result._
 import rps.model.error._
-import rps.repository.{InMemoryGameRepository}
+import rps.repository.{GameRepository}
 
-case class GameService(implicit repo: InMemoryGameRepository) {
+trait GameService {
+  def playGame(move: Move): Unit;
+  def getGame(): Option[GameSummary];
+}
+
+case class GameServiceImpl(implicit repo: GameRepository) extends GameService {
```

Comment:
I would make `repo` explicit, so that the logical wiring in Boot is clearer to read.

Reply:
is there a general rule of thumb of when it is a good idea to use implicit parameters?

Reply:
It's a general tradeoff between being explicit and being verbose.

In this case, you're defining a value once and you're passing it once: the code saves is neglegible and the clear dependency is lost in the implicit scope.

With a logger or an execution context, it's different story: you're passing a value everywhere and it doesn't add much to the clarity of the story. You don't care that much that a logger is being passed around, right?